src/inout.cpp

```diff
@@ -0,0 +1,61 @@
+#include "inout.h"
+#include "mongoose.h"
+
+namespace graft
+{
+std::unordered_map<std::string, std::string> OutHttp::uri_substitutions;
+
+//following #define depends on hm such as const http_message&
+#define SET_STR_FLD(fld) \
```

Comment:
I see no stringnification or other tricks here... so is there a good reason to keep it a macros vs. making it a inline function?

Reply:
There were tricks in my previous attempts, when it held mg_str like structures and set the same fields from objects of different types. But now it is not required. Have fixed.